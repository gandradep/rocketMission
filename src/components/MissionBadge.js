import React from 'react';
import PropTypes from 'prop-types';
import classes from './MissionButton.module.css';

const MissionBadge = (props) => {
  const { reserved } = props;
  if (reserved) {
    return (
      <button
        type="button"
        className={`${classes.badge} ${classes.bgBlue}`}
      >
        Active Member
      </button>
    );
  }
  return (
    <button
      type="button"
      className={`${classes.badge} ${classes.bgGray}`}
    >
      NOT A MEMBER
    </button>
  );
};
MissionBadge.propTypes = {
  reserved: PropTypes.bool.isRequired,
};

export default MissionBadge;
