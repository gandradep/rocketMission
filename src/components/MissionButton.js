import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import classes from './MissionButton.module.css';
import { joinMission } from '../redux/missions/missions';

const MissionButton = (props) => {
  const { id, reserved } = props;
  const dispatch = useDispatch();
  const joinMissionHandler = (e) => {
    dispatch(joinMission(e.target.id));
  };
  if (reserved) {
    return (
      <div>
        <button
          id={id}
          type="button"
          className={`${classes.button} ${classes.red}`}
          onClick={joinMissionHandler}
        >
          Leave Mission
        </button>
      </div>
    );
  }
  return (
    <div>
      <button
        id={id}
        type="button"
        className={`${classes.button} ${classes.gray}`}
        onClick={joinMissionHandler}
      >
        Join Mission
      </button>
    </div>
  );
};

MissionButton.propTypes = {
  id: PropTypes.string.isRequired,
  reserved: PropTypes.bool.isRequired,
};

export default MissionButton;
