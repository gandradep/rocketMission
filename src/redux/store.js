import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missions';
import rocketReducer from './rocketAction';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionsReducer,
  },
});

export default store;
