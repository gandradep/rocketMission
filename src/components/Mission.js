import React from 'react';
import PropTypes from 'prop-types';
import MissionBadge from './MissionBadge';
import MissionButton from './MissionButton';
import classes from './Missions.module.css';

const Mission = (props) => {
  const { name, description } = props;
  return (
    <tr>
      <td
        className={`${classes.border} ${classes.bold} ${classes.verticalTop}`}
      >
        {name}
      </td>
      <td className={classes.border}>{description}</td>
      <td className={`${classes.border} ${classes.center}`}>
        <MissionBadge />
      </td>
      <td className={`${classes.border} ${classes.center}`}>
        <MissionButton />
      </td>
    </tr>
  );
};
Mission.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default Mission;
