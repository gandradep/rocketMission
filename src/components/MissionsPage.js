import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissions } from '../redux/missions/missions';
import classes from './Missions.module.css';
import Mission from './Mission';

let fetchOnce = true;

const MissionsPage = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions);

  useEffect(() => {
    if (fetchOnce) {
      dispatch(getMissions());
      fetchOnce = false;
    }
  }, [dispatch]);

  return (
    <div>
      <table
        className={`${classes.table} w-full `}
      >
        <thead>
          <tr id="rowTitle">
            <th className={`${classes.w15} ${classes.border} ${classes.textLeft}`}>Mission</th>
            <th className={`${classes.w55} ${classes.border} ${classes.textLeft}`}>Description</th>
            <th className={`${classes.w15} ${classes.border} ${classes.textLeft}`}>Status</th>
            <th className={`${classes.w15} ${classes.border} ${classes.colorWhite}`}>.</th>
          </tr>
        </thead>
        <tbody>
          {missions?.map((mission) => (
            <Mission
              key={mission.id}
              id={mission.id}
              name={mission.name}
              description={mission.description}
            />
          ))}
        </tbody>

      </table>
    </div>
  );
};

export default MissionsPage;
