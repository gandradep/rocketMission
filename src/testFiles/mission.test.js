import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Mission from '../components/Mission';
import MissionsPage from '../components/MissionsPage';
import MissionBadge from '../components/MissionBadge';
import MissionButton from '../components/MissionButton';

describe('Missions components', () => {
  it('MissionsPage renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <MissionsPage />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Mission renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <Mission />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('MissionBadge renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <MissionBadge />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('MissionButton renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <MissionButton />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
