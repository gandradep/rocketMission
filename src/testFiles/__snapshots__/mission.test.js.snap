// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions components Mission renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr>
        <td
          class="border bold verticalTop"
        />
        <td
          class="border"
        />
        <td
          class="border center"
        >
          <button
            class="badge bgGray"
            type="button"
          >
            NOT A MEMBER
          </button>
        </td>
        <td
          class="border center"
        >
          <div>
            <button
              class="button gray"
              type="button"
            >
              Join Mission
            </button>
          </div>
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr>
      <td
        class="border bold verticalTop"
      />
      <td
        class="border"
      />
      <td
        class="border center"
      >
        <button
          class="badge bgGray"
          type="button"
        >
          NOT A MEMBER
        </button>
      </td>
      <td
        class="border center"
      >
        <div>
          <button
            class="button gray"
            type="button"
          >
            Join Mission
          </button>
        </div>
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Missions components MissionBadge renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="badge bgGray"
        type="button"
      >
        NOT A MEMBER
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="badge bgGray"
      type="button"
    >
      NOT A MEMBER
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Missions components MissionButton renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <button
          class="button gray"
          type="button"
        >
          Join Mission
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        class="button gray"
        type="button"
      >
        Join Mission
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Missions components MissionsPage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <table
          class="table w-full "
        >
          <thead>
            <tr
              id="rowTitle"
            >
              <th
                class="w15 border textLeft"
              >
                Mission
              </th>
              <th
                class="w55 border textLeft"
              >
                Description
              </th>
              <th
                class="w15 border textLeft"
              >
                Status
              </th>
              <th
                class="w15 border colorWhite"
              >
                .
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <table
        class="table w-full "
      >
        <thead>
          <tr
            id="rowTitle"
          >
            <th
              class="w15 border textLeft"
            >
              Mission
            </th>
            <th
              class="w55 border textLeft"
            >
              Description
            </th>
            <th
              class="w15 border textLeft"
            >
              Status
            </th>
            <th
              class="w15 border colorWhite"
            >
              .
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
